components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIStoreHostObject.java

```diff
@@ -1599,6 +1600,30 @@ public static boolean jsFunction_addAPISubscription(Context cx,
             trustedIdp = args[7].toString();
         }
 
+        APIConsumer apiConsumer = getAPIConsumer(thisObj);
+        Subscriber subscriber = apiConsumer.getSubscriber(userId);
+        if (subscriber == null) {
+            subscriber = new Subscriber(userId);
+            subscriber.setSubscribedDate(new Date());
+            //TODO : need to set the proper email
+            subscriber.setEmail("");
+            try {
+                int tenantId =
+                        ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                                              .getTenantId(
+                                                      MultitenantUtils.getTenantDomain(userId));
+                subscriber.setTenantId(tenantId);
+                apiConsumer.addSubscriber(subscriber);
+            } catch (AppManagementException e) {
+                handleException("Error while adding the subscriber" + subscriber.getName(), e);
+                return false;
```

Comment:
handleException method throws a new exception. Hence having return statement after that make no sense.